src/main/java/com/prgrms/vouchermanager/repository/voucher/VoucherMemoryRepository.java

```diff
@@ -0,0 +1,57 @@
+package com.prgrms.vouchermanager.repository.voucher;
+
+import com.prgrms.vouchermanager.domain.voucher.FixedAmountVoucher;
+import com.prgrms.vouchermanager.domain.voucher.Voucher;
+import com.prgrms.vouchermanager.domain.voucher.VoucherFactory;
+import com.prgrms.vouchermanager.domain.voucher.VoucherType;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Repository;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+@Repository
+@Profile("dev")
+public class VoucherMemoryRepository implements VoucherRepository {
+
+    private final Map<UUID, Voucher> vouchers = new HashMap<>();
+
+    public Voucher create(Voucher voucher) {
+
+        vouchers.put(voucher.getId(), voucher);
+        return voucher;
+    }
+
+    public List<Voucher> list() {
+        return vouchers
+                .values()
+                .stream()
+                .toList();
+    }
+
+    @Override
+    public Voucher findById(UUID id) {
+        return vouchers.get(id);
+    }
+
+    @Override
+    public Voucher updateDiscount(UUID id, int discount) {
+        Voucher voucher = vouchers.get(id);
+        vouchers.remove(id);
+
+        Voucher updateVoucher = VoucherFactory.update(id,
+                        voucher instanceof FixedAmountVoucher ? VoucherType.FIXED : VoucherType.PERCENT,
+                        discount)
+                .get();
+        vouchers.put(id, updateVoucher);
+        return updateVoucher;
+    }
```

Comment:
```suggestion
    public Voucher updateDiscount(UUID id, int discount) {
        Voucher voucher = vouchers.computeIfPresent(id, 
                (k, v) -> VoucherFactory.update(id,
                        v instanceof FixedAmountVoucher ? VoucherType.FIXED : VoucherType.PERCENT,
                        discount)
                .get());

        fileIO.updateFile(vouchers);
        return voucher;
    }
```
Map 에는 computeIfPresent api가 있습니다. delete  -> put 하지 않아도 되지 않을까 싶습니다.!